src/sharding/ShardingManager.js

```diff
@@ -111,12 +136,15 @@ class ShardingManager extends EventEmitter {
   async spawn(amount = this.totalShards, delay = 5500, waitForReady = true) {
```

Comment:
Rather than do your magic shit with start and end shards in the constructor options, my personal preference would be to see this `amount` parameter accept an array of shard IDs or at least some way of specifying the range here as opposed to the constructor.